tests/server/test_handlers.py

```diff
@@ -78,6 +81,78 @@ def test_search_runs_default_view_type(mock_get_request_message, mock_store):
     assert args[2] == ViewType.ACTIVE_ONLY
 
 
+def _assert_logged_entities(run_id, metric_entities, param_entities, tag_entities):
+    client = mlflow.tracking.MlflowClient()
+    store = mlflow.tracking.utils._get_store()
+    run = client.get_run(run_id)
+    # Assert logged metrics
+    all_logged_metrics = sum([store.get_metric_history(run_id, m.key)
+                              for m in run.data.metrics], [])
+    assert len(all_logged_metrics) == len(metric_entities)
+    logged_metrics_dicts = [dict(m) for m in all_logged_metrics]
+    for metric in metric_entities:
+        assert dict(metric) in logged_metrics_dicts
+    # Assert logged params
+    param_entities_dict = [dict(p) for p in param_entities]
+    for p in run.data.params:
+        assert dict(p) in param_entities_dict
+    # Assert logged tags
+    tag_entities_dict = [dict(t) for t in tag_entities]
+    approx_expected_tags = [MLFLOW_SOURCE_NAME, MLFLOW_SOURCE_TYPE]
+    for t in run.data.tags:
+        if t.key in approx_expected_tags:
+            pass
+        else:
+            assert dict(t) in tag_entities_dict
+
+
+def test_log_batch_handler_success(mock_get_request_message, tmpdir):
```

Comment:
this is really an awesome test -- at some point we can use the `@pytest.mark.parametrize` to send it different stores.